src/main/java/org/cyclonedx/util/deserializer/AffectDeserializer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of CycloneDX Core (Java).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.util.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.cyclonedx.model.vulnerability.Vulnerability;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+public class AffectDeserializer
```

Comment:
JOOC Why is this other serializer needed? is not done automatically with the VulnerabilitySerializer?

Reply:
Same question for the list

Reply:
> JOOC Why is this other serializer needed? is not done automatically with the VulnerabilitySerializer?

It wasn't done automatically. Without the affects deserializer the following [test ](https://github.com/shaikhu/cyclonedx-core-java/blob/691649c4d0f7adeba4a27ddb1ddf33a69f0c9c1b/src/test/java/org/cyclonedx/parsers/AbstractParserTest.java#L597)failed.

Reply:
> Same question for the list

See above comment.